ninja_simple_jwt/auth/backend.py

```diff
@@ -0,0 +1,86 @@
+from django.contrib.auth.base_user import AbstractBaseUser
+from django.http import HttpRequest
+from jwt import PyJWTError
+
+from ninja_simple_jwt.jwt.token_operations import TokenTypes, decode_token
+from ninja_simple_jwt.settings import ninja_simple_jwt_settings
+
+User: Type[AbstractBaseUser] = get_user_model()  # type: ignore[assignment]
+
+
+class JwtBackend(BaseBackend):
+    def authenticate(
+        self, request: Optional[HttpRequest], token: Optional[str] = None, **kwargs: Any
+    ) -> Optional[AbstractBaseUser]:
+        """
+        Authenticate a user using a JWT token.
+
+        Args:
+            request: The HTTP request object
+            token: The JWT token string
+
+        Returns:
+            User instance if authentication successful, None otherwise
+        """
+        if not token:
+            return None
+
+        try:
+            # Decode and verify the JWT token
+            decoded_token = decode_token(token, token_type=TokenTypes.ACCESS, verify=True)
+
+            # Extract user_id from the token
+            user_id = decoded_token.get("user_id")
+            if not user_id:
+                return None
+
+            # Create a stateless user instance from token claims
+            user = self._create_user_from_token(decoded_token)
+            return user
+
+        except PyJWTError:
+            # Token is invalid, expired, or malformed
+            return None
+
+    @staticmethod
+    def _create_user_from_token(token: dict) -> AbstractBaseUser:
+        """
+        Create a user instance from JWT token claims without database access.
+
+        This creates a "fake" User instance that Django ORM will accept for queries
+        by properly setting its primary key and state.
+
+        Args:
+            token: The decoded JWT token payload
+
+        Returns:
+            User instance populated with token claims
+        """
+        user = User()
+
+        # Set the primary key based on user_id from token
+        user_id = token.get("user_id")
+
+        # Find the primary key field name (could be 'id', 'pk', 'guid', etc.)
+        pk_field = User._meta.pk.name  # type: ignore[union-attr]
+        setattr(user, pk_field, user_id)
+
+        # Mark the instance as existing in the database (critical for ORM queries)
+        user._state.adding = False  # pylint: disable=protected-access
+
+        # Set attributes from token claims
+        for claim, user_attribute in ninja_simple_jwt_settings.TOKEN_CLAIM_USER_ATTRIBUTE_MAP.items():
+            if claim in token:
+                if isinstance(user_attribute, str):
+                    setattr(user, user_attribute, token.get(claim))
+                else:
+                    # Handle callable user attributes (like lambda functions)
+                    # For callables, we can't call them without a real user object,
+                    # so we set the claim directly as an attribute
+                    setattr(user, claim, token.get(claim))
```

Comment:
I don't think I understand this section. Is `TOKEN_CLAIM_USER_ATTRIBUTE_MAP` a `dict[str, str]`, or does it support `dict[str, str | Callable]`? Or are you checking to make sure that the attribute on the `User` type is a callable?